pkg/kubelet/kuberuntime/kuberuntime_manager.go

```diff
@@ -615,13 +626,17 @@ func (m *kubeGenericRuntimeManager) computePodResizeAction(pod *v1.Pod, containe
 			currentContainerResources: &currentResources,
 		}
 		// Order the container updates such that resource decreases are applied before increases
```

Comment:
I think both regular containers and restartable init containers whose resources are decreased should be resized earlier so that the sum of containers' limits doesn't exceed the pod's limit while resizing. It might be better to put all containers that will be resized into `ContainersToUpdate`.